src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -26,22 +26,22 @@
     @DataProvider(name = "loadReadsBAM")
     public Object[][] loadReadsBAM() {
         return new Object[][]{
-                {testDataDir + "engine/dataflow/ReadsSource/HiSeq.1mb.1RG.2k_lines.bam", "ReadsSparkSinkUnitTest1.bam"},
-                {testDataDir + "tools/BQSR/expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam", "ReadsSparkSinkUnitTest2.bam"},
+                {testDataDir + "engine/dataflow/ReadsSource/HiSeq.1mb.1RG.2k_lines.bam", createTempFile("HiSeq.1mb.1RG.2k_lines", ".bam").getAbsolutePath()},
+                {testDataDir + "tools/BQSR/expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam", createTempFile("HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate", ".bam").getAbsolutePath()},
         };
     }
 
     @DataProvider(name = "loadReadsADAM")
     public Object[][] loadReadsADAM() {
         return new Object[][]{
-                {testDataDir + "engine/dataflow/ReadsSource/HiSeq.1mb.1RG.2k_lines.bam", "ReadsSparkSinkUnitTest1_ADAM"},
-                {testDataDir + "tools/BQSR/expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam", "ReadsSparkSinkUnitTest2_ADAM"},
+                {testDataDir + "engine/dataflow/ReadsSource/HiSeq.1mb.1RG.2k_lines.bam", createTempFile("HiSeq.1mb.1RG.2k_lines", ".adam").getAbsolutePath()},
+                {testDataDir + "tools/BQSR/expected.HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate.bam", createTempFile("HiSeq.1mb.1RG.2k_lines.bqsr.DIQ.alternate", ".adam").getAbsolutePath()},
         };
     }
 
     @Test(dataProvider = "loadReadsBAM", groups = "spark")
     public void readsSinkTest(String inputBam, String outputFile) throws IOException {
-        new File(outputFile).deleteOnExit();
+        new File(outputFile).delete();
```

Comment:
Would `IOUtils.deleteRecursivelyOnExit()` help here?  It would be best if the test doesn't leave any output when it's done.